internal/cli/status.go

```diff
@@ -0,0 +1,830 @@
+		},
+		Workspace: &pb.Ref_Workspace{
+			Workspace: workspace,
+		},
+		LoadDetails: pb.Deployment_BUILD,
+	})
+	if err != nil {
+		return err
+	}
+
+	deployHeaders := []string{
+		"App Name", "Version", "Physical State", "Id", "Artifact Id", "Exec", "Logs",
+	}
+
+	deployTbl := terminal.NewTable(deployHeaders...)
+
+	resourcesHeaders := []string{
+		"Type", "Platform", "Category",
+	}
+
+	resourcesTbl := terminal.NewTable(resourcesHeaders...)
+
+	if len(respDeployList.Deployments) > 0 {
+		deploy := respDeployList.Deployments[0]
+		statusColor := ""
+
+		columns := []string{
+			deploy.Application.Application,
+			fmt.Sprintf("v%d", deploy.Sequence),
+			deploy.Status.State.String(),
+			deploy.Id,
+			deploy.ArtifactId,
+			strconv.FormatBool(deploy.HasExecPlugin),
+			strconv.FormatBool(deploy.HasLogsPlugin),
+		}
+
+		// Add column data to table
+		deployTbl.Rich(
+			columns,
+			[]string{
+				statusColor,
+			},
+		)
+
+		// Deployment Resources Summary
+		//   Resources List
+		for _, dr := range deploy.DeclaredResources {
+			columns := []string{
+				dr.Name,
+				dr.Platform,
+				dr.CategoryDisplayHint.String(),
+			}
+
+			// Add column data to table
+			resourcesTbl.Rich(
+				columns,
+				[]string{
+					statusColor,
+				},
+			)
+		}
+
+	} // else show no table
+
+	// TODO(briancain): we don't yet store a list of recent events per app
+	// but it would go here if we did.
+	// Recent Events
+	//   Events List
+
+	if c.flagJson {
+		c.outputJsonAppStatus(appTbl, deployTbl, resourcesTbl, project)
+	} else {
+		c.ui.Output("")
+		c.ui.Output("Application Summary")
+		c.ui.Table(appTbl, terminal.WithStyle("Simple"))
+		c.ui.Output("")
+		c.ui.Output("Deployment Summary")
+		c.ui.Table(deployTbl, terminal.WithStyle("Simple"))
+		c.ui.Output("")
+		c.ui.Output("Deployment Resources Summary")
+		c.ui.Table(resourcesTbl, terminal.WithStyle("Simple"))
+		c.ui.Output("")
+		c.ui.Output(wpStatusAppSuccessMsg)
+	}
+
+	if appFailures {
+		c.ui.Output("")
+
+		c.ui.Output(wpStatusHealthTriageMsg, projectTarget, terminal.WithWarningStyle())
+	}
+
+	return nil
+}
+
+// FormatProjectStatus formats all known projects into a table
+func (c *StatusCommand) FormatProjectStatus() error {
+	if !c.flagJson {
+		c.ui.Output(wpStatusMsg, c.serverCtx.Server.Address)
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	projectResp, err := client.ListProjects(c.Ctx, &empty.Empty{})
+	if err != nil {
+		c.ui.Output("Failed to retrieve all projects", terminal.WithErrorStyle())
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return err
+	}
+	projNameList := projectResp.Projects
+
+	headers := []string{
+		"Project", "Workspace", "App Statuses",
+	}
+
+	tbl := terminal.NewTable(headers...)
+
+	for _, projectRef := range projNameList {
+		resp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+			Project: projectRef,
+		})
+		if err != nil {
+			return err
+		}
+
+		var workspace string
+		if len(resp.Workspaces) != 0 {
```

Comment:
This part is repeated a couple of times. I guess it's a good candidate for a function